### ISSUE TITLE
403 Must access using HTTPS instead of HTTP

### ISSUE THREAD

**Issue by mtrienis** [astraw/stdeb#122]
Hello, 

I'm trying to run:

pypi-install diamond

However, I'm getting the following error. 

```
             Traceback (most recent call last):
                File "/usr/bin/pypi-install", line 142, in <module>
                  main()
                File "/usr/bin/pypi-install", line 110, in main
                  tarball_fname = get_source_tarball(package_name,verbose=options.verbose)
                File "/usr/bin/pypi-install", line 61, in get_source_tarball
                  verbose=verbose)
                File "/usr/bin/pypi-install", line 30, in find_tar_gz
                  releases = pypi.package_releases(package_name)
                File "/usr/lib/python2.7/xmlrpclib.py", line 1233, in __call__
                  return self.__send(self.__name, args)
                File "/usr/lib/python2.7/xmlrpclib.py", line 1587, in __request
                  verbose=self.__verbose
                File "/usr/lib/python2.7/xmlrpclib.py", line 1273, in request
                  return self.single_request(host, handler, request_body, verbose)
                File "/usr/lib/python2.7/xmlrpclib.py", line 1321, in single_request
                  response.msg,
              xmlrpclib.ProtocolError: <ProtocolError for pypi.python.org/pypi: 403 Must access using HTTPS instead of HTTP>
```

It actually fails in the same way for any package, not just diamond. 

I'm currently running:

`Version: 0.6.0+20100620-4ubuntu1`

**Comment by SteveAyre** [comment]
The issue appears to be stdeb's usage of xmlrpclib. The transport it creates and passes to xmlrpclib.ServerProxy to set the User-Agent does not support HTTPS. If transport is omitted xmlrpclib would automatically pick an internal HTTP or HTTPS transport, but stdeb overrides this.

The proper fix would be to configure the transport to support HTTPS.

A workaround is to edit pypi-install so that the transport parameter is not provided to xmlrpclib.ServerProxy so it uses the default.

**Comment by dirk-thomas** [comment]
This has been addressed as of version 0.8.0 (see 60bea01bd6c69ce0c7bf31952604ad70567a2e76) which is already 4.5 years old. Therefore I will go ahead and close this issue.
